libraries/AP_OSD/AP_OSD_Screen.cpp

```diff
@@ -1623,7 +1623,9 @@ void AP_OSD_Screen::draw_blh_rpm(uint8_t x, uint8_t y)
     if (!AP::esc_telem().get_rpm(0, rpm)) {
         return;
     }
-    backend->write(x, y, false, "%3.1f%c%c", uint16_t(rpm) * 0.001f, SYM_KILO, SYM_RPM);
+    float krpm = rpm * 0.001f;
+    const char *format = krpm < 9.95 ? "%.2f%c%c" : (krpm < 99.95 ? "%.1f%c%c" : "%.0f%c%c");
```

Comment:
Why not 10 and 100?

Reply:
Because it wouldn't switch the format at the right moment. 9.95 %.1f displays 10.0